packages/roosterjs-editor-plugins/lib/plugins/HyperLink/HyperLink.ts

```diff
@@ -67,6 +76,43 @@ export default class HyperLink implements EditorPlugin {
      * @param event PluginEvent object
      */
     public onPluginEvent(event: PluginEvent): void {
+        if (
+            event.eventType == PluginEventType.MouseUp ||
+            (event.eventType == PluginEventType.KeyUp &&
+                event.rawEvent.which >= Keys.PAGEUP &&
+                event.rawEvent.which <= Keys.DOWN) ||
+            event.eventType == PluginEventType.ContentChanged
+        ) {
+            const anchor = this.editor.getElementAtCursor(
+                'A[href]',
+                null /*startFrom*/,
+                event
+            ) as HTMLAnchorElement;
+
+            // If cursor has moved out of previously tracked link
+            // update link href if applicable and then reset tracked state
+            if (this.trackedLink && anchor !== this.trackedLink) {
+                if (!this.doesLinkDisplayMatchHref(this.trackedLink)) {
+                    this.updateLinkHref(event);
+                }
+
+                this.trackedLink = null;
+                this.originalHref = '';
+            }
+
+            // If the link's href value was edited, stop tracking the link.
+            if (this.trackedLink && this.tryGetHref(this.trackedLink) !== this.originalHref) {
+                this.trackedLink = null;
+                this.originalHref = '';
+            }
+
+            // Cache link and href value if its href attribute currently matches its display text
+            if (!this.trackedLink && this.doesLinkDisplayMatchHref(anchor)) {
+                this.trackedLink = anchor;
+                this.originalHref = this.tryGetHref(anchor);
+            }
+        }
```

Comment:
The issue that link can't be updated  before click away can still happen.

Suggest always update the link for any key input or content changed event as long as this.trackedLink is not null.

Reply:
Chatted offline to resolve this. Summary is that one of the issues with changing behavior to update the link on any key input is that it makes it harder for the changes to be undo-able. @JiuqingSong suggested that the behavior kicks in on clicking away, handling any non-content input, and on blur. The latest commit has these updates.

Reply:
I also added handling for the space key as you could edit a link at the end and press space to leave the link element